src/TensorNetwork.jl

```diff
@@ -53,25 +55,43 @@ struct TensorNetwork <: AbstractTensorNetwork
             dict
         end
 
-        # Check for inconsistent index dimensions
-        for ind in keys(indexmap)
-            dims = map(tensor -> size(tensor, ind), indexmap[ind])
-            length(unique(dims)) == 1 || throw(DimensionMismatch("Index $(ind) has inconsistent dimension: $(dims)"))
+        if isnothing(unsafe) # If there is not an active UnsafeContext, check for index sizes
+            for ind in keys(indexmap) # Check for inconsistent index dimensions
+                dims = map(tensor -> size(tensor, ind), indexmap[ind])
+                length(unique(dims)) == 1 ||
+                    throw(DimensionMismatch("Index $(ind) has inconsistent dimension: $(dims)"))
+            end
+
+            return new(indexmap, tensormap, CachedField{Vector{Tensor}}(), Ref{Union{Nothing,UnsafeContext}}(nothing))
         end
 
-        return new(indexmap, tensormap, CachedField{Vector{Tensor}}())
+        return new(indexmap, tensormap, CachedField{Vector{Tensor}}(), Ref{Union{Nothing,UnsafeContext}}(unsafe))
     end
 end
 
 TensorNetwork() = TensorNetwork(Tensor[])
 TensorNetwork(tn::TensorNetwork) = tn
 
+function set_unsafe_context!(tn::TensorNetwork, uc::Union{Nothing,UnsafeContext})
+    tn.unsafe[] = uc
+    return tn
+end
```

Comment:
Can you generalize this to `AbstractTensorNetwork`?
```suggestion
function set_unsafe_context!(tn::AbstractTensorNetwork, uc::Union{Nothing,UnsafeContext})
    TensorNetwork(tn).unsafe[] = uc
    return tn
end
```